llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -1828,8 +1857,17 @@ struct DSEState {
         // underlying objects is very uncommon. If it turns out to be important,
         // we can use getUnderlyingObjects here instead.
         const Value *UO = getUnderlyingObject(DefLoc->Ptr);
-        if (!isInvisibleToCallerAfterRet(UO))
-          continue;
+        if (!isInvisibleToCallerAfterRet(UO)) {
+          // EVM local begin
+          Triple TT(DefI->getFunction()->getParent()->getTargetTriple());
+          unsigned LocAS = DefLoc->Ptr->getType()->getPointerAddressSpace();
+          if (!TT.isEVM() || LocAS == 5 || LocAS == 6)
```

Comment:
Can we use something instead of magic numbers?

Reply:
Thank you for pointing that out. I'll remove the address space check, as it's no longer necessary. The EVMSystemIntrinsicsPromotion pass now adds ptr addrspace(5) null, ptr addrspace(6) null parameters to system revert/return intrinsics, which ensures proper dependency tracking between stores and those intrinsics.